### PR TITLE
prepare for migration to GitHub merge queue

### PR DESCRIPTION
**Pull request by alaviss**
## Summary
Prepare workflows and documentations for migrating from bors-ng to GitHub Merge Queue.

## Details
Other than some reformatting here and there, the main changes centers
around making our workflows aware of merge_group event and to make
them pass on PRs even when they're skipped.

For publisher, it will now operate under an environment to allow for
stricter deployment rules and signed releases in the future.

Documentation has also been updated, and a new /merge command has been
added for trimming PR description and queuing a merge.

---

## Notes for Reviewers
* This PR template will not work well with Merge Queue since it doesn't support cutting off at a given marker.
* Testing shows that comments in markdown are also kept.

### Todo list
- [x] Prepare workflows for migration
- [x] Update CI documentation
~~- Update PR template~~
- [x] Add /merge command that auto trim PR description

### CODE REVIEW

**Comment by saem** [comment]
bors cancel

**Comment by saem** [comment]
OK, time to start updating docs or something...

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,6 +24,8 @@ on:
       - reopened
       - ready_for_review
 
+  merge_group:
```
Comment (saem) [inline .github/workflows/ci.yml:27]:
```suggestion
  merge_group:
    # build and tests all additions to the merge queue
```

File: .github/workflows/merge-pr.yml [hunk 2]
```diff
@@ -0,0 +1,135 @@
+name: /merge handler
```
Comment (saem) [inline .github/workflows/merge-pr.yml]: maybe this is a better file name `slash-command-merge.yml`?

File: .github/workflows/merge-pr.yml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+            if (pr_info.autoMergeRequest || pr_info.mergeQueueEntry) {
+              core.notice('PR is already queued for merging.');
+              return;
+            }
+
+            let [ body, body_tail ] = pr_info.body.split(/^---\s*$/m, 2);
+            body = body.trimEnd();
+            body_tail = body_tail?.trimEnd() || '';
+            core.setOutput('run', 'true');
```
Comment (saem) [inline .github/workflows/merge-pr.yml]:
```suggestion
            core.setOutput('run', 'true'); // prevents data loss, see `find-comment` step below
```

**Review by saem — APPROVED** [review]
few mild suggestions, but looks good to me.

**Comment by alaviss** [comment]
bors r+

Let's go

**Pull request merged**